rasa_core/training/structures.py

```diff
@@ -235,78 +235,76 @@ def with_cycles_removed(self):
         # type: () -> StoryGraph
         """Create a graph with the cyclic edges removed from this graph."""
 
-        if not self.cyclic_edge_ids:
-            return self
```

Comment:
i don't think we should be getting rid of this?

Reply:
this is the main thing, `OR` checkpoints should be eliminated even if there are no cycles, so we need to call `_remove_unused_generated_cps`, so I inverted this `if` instead of calling `_remove_unused_generated_cps` two times